### PR TITLE
Add a new webpush test that utilizes the new web push standard with userAuth

### PR DESCRIPTION
**Pull request by nojunpark**

### CODE REVIEW

File: test/integration/test/new_webpush_test.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        }
+      }
+      expect(pushService).equals(true);
+    });
+  });
+
+  it('Add push subscription',function(){     
+    var pushURI = Prepper.webPush_server.getEndpointURI();
+    var pushkey = Prepper.webPush_server.getPublicKey();
+    var userAuth = Prepper.webPush_server.getUserAuth();
```
Comment (JohanLorenzo) [inline test/integration/test/new_webpush_test.js]: It seems to be the only line added (compared to `test/integration/test/webpush_test.js`). I think we can have one single test files that deals with with both old and new webpush. For example:

``` js
Prepper.makeSuite('Test Push Service locally',function(){

  var baseSubscriptionPayload = {
     'select': {
       'id':setter
     }, 
     'value': {
       'Json': {
         'subscriptions':[{
           'push_uri': Prepper.webPush_server.getEndpointURI(),
           'public_key': Prepper.webPush_server.getPublicKey()
         }]
      }
    }

  var newWebPushSubscriptionPayload = Object.assign({}, baseSubscriptionPayload, {
    value: {
       Json: {
          subscriptions: {
            auth: Prepper.webPush_server.getUserAuth();
          }
       }
    }
  });

   var testParams = [{
      suiteName: 'Old WebPush',
      payload: baseSubscriptionPayload
    }, {
      suiteName: 'New WebPush',
      payload: newWebPushSubscriptionPayload
    }];

   testParams.forEach(testParam => {
      describe(testParam.suiteName, function() {
         // do all the tests here.
      });
   })
  };
});
```

**Comment by JohanLorenzo** [comment]
~~r? @JohanLorenzo~~

The server-side looks good, but I would avoid to copy the whole test suite. It looks hardly maintainable, especially knowing that there's only 1 tiny difference between the 2 suites. I left [a proposal](https://github.com/fxbox/foxbox/pull/428/files#r61105317) to get rid of the duplication.

**Comment by nojunpark** [comment]
r?  Added the check to delete the subscription as well. (necessary to run the test twice)

File: test/integration/test/webpush_test.js [hunk 2]
```diff
@@ -5,127 +5,187 @@ const chakram = require('chakram'), expect = chakram.expect;
...
+    }
+  };
+
+  var newWebPushSubscriptionPayload = 
+  Object.assign({}, baseSubscriptionPayload, {
+    value: {
+       Json: {
+          subscriptions: [{
+            'push_uri': Prepper.webPush_server.getEndpointURI(),
+           'public_key': Prepper.webPush_server.getPublicKey(),
```
Comment (JohanLorenzo) [inline test/integration/test/webpush_test.js]: Nit: Indentation

File: test/integration/test/webpush_test.js [hunk 3]
```diff
@@ -5,127 +5,187 @@ const chakram = require('chakram'), expect = chakram.expect;
...
+    }
+  });
+
+  var testParams = [{
+    suiteName: 'Old WebPush',
+    payload: baseSubscriptionPayload
+  }, {
+    suiteName: 'New WebPush',
+    payload: newWebPushSubscriptionPayload
+    }
```
Comment (JohanLorenzo) [inline test/integration/test/webpush_test.js]: Nit: Indentation

**Comment by JohanLorenzo** [comment]
Looks good to me. r+

**Pull request merged**